src/feat/online-feature.cc

```diff
@@ -24,18 +24,55 @@
 
 namespace kaldi {
 
+RecyclingVector::RecyclingVector(int items_to_hold) : items_to_hold_(items_to_hold),
+                                                  first_available_index_(0) { }
```

Comment:
Maybe add a `KALDI_ASSERT(items_to_hold > 0);`. The behavior would be fascinating if this is zero. : )

Reply:
+1 - eagle eyes ;)

Reply:
I've actually mapped 0 to -1, hopefully making it more user-friendly than program termination.